### PR TITLE
[Review] `akka-http` clients for CouchDB and Consul

### PR DESCRIPTION
**Pull request by psuter**
@rabbah @markusthoemmes for your reading pleasure

### CODE REVIEW

**Comment by psuter** [comment]
Note that the commits aren't in order because [GitHub just doesn't like rebases](https://help.github.com/articles/why-are-my-commits-in-the-wrong-order/). The last three commits are orthogonal changes, though, so it shouldn't affect reviewability.

File: common/scala/src/main/resources/application.conf [hunk 1]
```diff
@@ -1,2 +1,12 @@
 # default application configuration file for spray/akka
-include "logging"
\ No newline at end of file
+include "logging"
+
+akka.http {
+    client {
+        parsing.illegal-header-warnings = off
```
Comment (psuter) [inline common/scala/src/main/resources/application.conf]: This gets rid of those annoying warnings that we get from CouchDB responses, btw.

**Comment by psuter** [comment]
PPG 684

**Comment by psuter** [comment]
Looks like this needs careful rebasing after `7454e01cf035053813690a619fb0f9ce163e4718`

File: common/scala/src/main/scala/whisk/common/ConsulClient.scala [hunk 2]
```diff
@@ -55,18 +53,16 @@ object ConsulEntry extends DefaultJsonProtocol {
  * Client to access Consul's <a href="https://www.consul.io/docs/agent/http/kv.html">
  * key/value store</a>
  */
-class ConsulClient(host: String)(implicit val actorSystem: ActorSystem) {
+class ConsulClient(hostAndPort: String)(implicit val actorSystem: ActorSystem) {
```
Comment (psuter) [inline common/scala/src/main/scala/whisk/common/ConsulClient.scala]: @ChrBi @markusthoemmes you may want to review this given your recent changes to `ConsulClient`

**Comment by psuter** [comment]
PPG 713 after latest rebase

**Comment by markusthoemmes** [comment]
`ConsulClient` LGTM

**Comment by rabbah** [comment]
good to merge.

**Comment by rabbah** [comment]
Close, given 824?

**Comment by psuter** [comment]
Included in #824

**Pull request closed without merging**